apps/frontend/src/routes/+layout.svelte

```diff
@@ -17,42 +22,74 @@
 		Dropdown,
 		DropdownHeader,
 		DropdownDivider,
-		Avatar,
-		DarkMode
+		Avatar
 	} from 'flowbite-svelte';
 
 	import LanguageSwitcher from '$lib/components/LanguageSwitcher.svelte';
 	import SignInButton from '$lib/auth/components/SignInButton.svelte';
 
 	let { children } = $props();
 
-	let isDarkMode = $state(true);
+	let isDarkMode = $state(false);
 
-	// Auto dark mode based on browser preference
+	// Initialize theme immediately
 	onMount(() => {
-		const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
-
-		const updateTheme = (e: MediaQueryList | MediaQueryListEvent) => {
-			if (e.matches) {
-				document.documentElement.classList.add('dark');
-				isDarkMode = true;
-			} else {
-				document.documentElement.classList.remove('dark');
-				isDarkMode = false;
-			}
-		};
+		// Check localStorage first, then fall back to system preference
+		const savedTheme = localStorage.getItem('theme');
+		const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+
+		isDarkMode = savedTheme === 'dark' || (!savedTheme && prefersDark);
+
+		if (isDarkMode) {
+			document.documentElement.classList.add('dark');
+		} else {
+			document.documentElement.classList.remove('dark');
+		}
```

Comment:
This pattern occurs several time in your patch.

I see (a combination of) two possible solutions:

1. Creating a function and using it both for the hook as well as calling it `onMount()`.
2. Using `$effect` to update class based on `isDarkMode`. Might require something like `$effect.pre` so we might avoid `onMount()`. 

I'd consider 1. good enough for now. 2. Nice to have.